src/main/java/mate/academy/bookstore/serivce/UserServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package mate.academy.bookstore.serivce;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.user.UserRegistrationRequestDto;
+import mate.academy.bookstore.dto.user.UserResponseDto;
+import mate.academy.bookstore.mapper.UserMapper;
+import mate.academy.bookstore.model.User;
+import mate.academy.bookstore.repository.UserRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```

Comment:
more efficient and simple
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```